modules/client/src/main/scala/MessagingAPI.scala

```diff
@@ -43,7 +43,8 @@ trait Consuming[F[_]] {
       noLocal: NoLocal = false,
       noAck: NoAck = true,
       exclusive: Boolean = false,
-      arguments: FieldTable = FieldTable.empty
+      arguments: FieldTable = FieldTable.empty,
+      consumerTag: ConsumerTag = ConsumerTag.random
```

Comment:
I think it's better to have an optional ctag with `None` default, which is the default incremental tags, while allowing users to pass a ctag when they want to override the name.

Reply:
Ok so `consumeRaw` should look like this 

```scala
  def consumeRaw(
      queue: QueueName,
      noLocal: NoLocal = false,
      noAck: NoAck = true,
      exclusive: Boolean = false,
      arguments: FieldTable = FieldTable.empty,
      consumerTag: Option[ConsumerTag] = None
  ): Stream[F, DeliveredMessageRaw]
```

Should  `ctag` in `final def consume` also be optional ?

Reply:
Yes, everywhere the `ctag` is added should be optional to maintain backward source compatibility.

Reply:
I made it optional in my last commit